### PR TITLE
Adds plugin list, search, and info support to the CLI

### PR DESCRIPTION
**Pull request by jamonholmgren**
Adds support for the following plugin commands:

```
ignite plugin list
ignite plugin search <search string>
ignite plugin info <plugin name>
```

Also adds a directory that we can add to as we find out about and test plugins.

screen shot 2017-02-22 at 10 12 56 pm

### CODE REVIEW

File: packages/ignite-cli/src/commands/plugin.js [hunk 1]
```diff
@@ -105,29 +106,110 @@ Generally, you would run this from ./YourApp/ignite/plugins/
 Commands:
   ignite plugin help
   ignite plugin new <your-plugin>
-  ignite plugin list (coming soon)
+  ignite plugin list
+  ignite plugin search <search string>
+  ignite plugin info <plugin name>
```
Comment (skellock) [inline packages/ignite-cli/src/commands/plugin.js]: I'd like to promote these to `ignite list`, `ignite search`, and `ignite info` just like we already have `ignite add` and `ignite remove`?

Ignite 2 is really all about the plugins.

File: packages/ignite-cli/src/commands/plugin.js [hunk 2]
```diff
@@ -105,29 +106,110 @@ Generally, you would run this from ./YourApp/ignite/plugins/
...
+  info(colors.highlight(`Ignite Plugin: '${name}'`))
+  newline()
+
+  if (plugin) {
+    table([
+      [ "Name", name ],
+      [ "Description", plugin.description ],
+      [ "Author", plugin.author ],
+      [ "URL", plugin.url ],
+      [ "Email", plugin.email ]
```
Comment (skellock) [inline packages/ignite-cli/src/commands/plugin.js]: I'd vote to drop this field.

File: packages/ignite-cli/src/commands/plugin.js [hunk 3]
```diff
@@ -105,29 +106,110 @@ Generally, you would run this from ./YourApp/ignite/plugins/
...
 
-  // context.print.debug(context.parameters)
   switch (parameters.second) {
     case 'new':
       await createNewPlugin(context)
       break
 
     case 'list':
-      print.error(`ignite plugin list is not implemented yet.`)
+      showPluginDirectory(context, null)
```
Comment (skellock) [inline packages/ignite-cli/src/commands/plugin.js]: If we do end up keeping the `plugin` namespace, could we move this to a gluegun plugin?  It handles this switch statement and organizes your code at the same time.

**Comment by jamonholmgren** [comment]
Good feedback @skellock -- we have some stuff to discuss on this. Good idea to bump back to 2.1 & let's table this until we've had a chance to chat about it.

File: packages/ignite-cli/src/commands/plugin.js (hunk 1 above)
Comment (franzejr) [inline packages/ignite-cli/src/commands/plugin.js]: I liked that!

**Comment by franzejr** [comment]
Just to mention, I really liked this idea of searching for packages. I think it's necessary!

File: packages/ignite-cli/src/commands/plugin.js (hunk 1 above)
Comment (maxfindel) [inline packages/ignite-cli/src/commands/plugin.js]: Wouldn't it be better to use a centralized server like npm or bundler? That way you don't have to push code every time someone creates a cool plugin. Who's up for the challenge?

Comment (skellock) [inline packages/ignite-cli/src/commands/plugin.js]: @ryanlntn is taking this task on now. 🎉

File: packages/ignite-cli/src/commands/info.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const validateName = require('../lib/validateName')
+const fetchPluginRegistry = require('./fetchPluginRegistry')
+
+/**
+ * Shows info about a particular plugin.
+ *
+ * @param {Object} context The gluegun context.
+ */
+const showPluginInfo = async function (context) {
+  console.log('I was called!')
```
Comment (skellock) [inline packages/ignite-cli/src/commands/info.js]: Hey, I just logged you.
And this is crazy!
So here's my console.
So cull me maybe?

**Comment by ryanlntn** [comment]
screen shot 2017-04-25 at 12 50 11 pm

I've moved the list, search, and info commands out of the plugin namespace and moved the plugin registry data to https://github.com/infinitered/ignite-plugins

screen shot 2017-04-25 at 12 53 43 pm
I'm thinking of simplifying list output to just the shortNames. Thoughts?

**Comment by skellock** [comment]
@ryanlntn I like the short names for displaying. ya.  👍

**Pull request merged**